src/rust/ide/src/controller/searcher.rs

```diff
@@ -209,6 +209,26 @@ impl ParsedInput {
             CompletedFragmentId::Argument {..}                  => TypingArgument,
         }
     }
+
+    /// Convert the current input to Prefix Chain representation.
+    pub fn as_prefix_chain
+    (&self, parser:&Parser) -> Option<ast::Shifted<ast::prefix::Chain>> {
+        let parsed_pattern = parser.parse_line(&self.pattern).ok();
+        let pattern_sast   = parsed_pattern.map(|p| ast::Shifted::new(self.pattern_offset,p));
+        if let Some(mut chain) = self.expression.clone() {
+            if let Some(sast) = pattern_sast {
+                let prefix_id = None;
+                let argument  = ast::prefix::Argument {sast,prefix_id};
+                chain.wrapped.args.push(argument);
+            }
+            Some(chain)
+        } else if let Some(sast) = pattern_sast{
```

Comment:
I would need a better explanation of the logic of these two if branches. It's not obvious to me why we fiorst pattern match on `self.expression` and if we fail, then we pattern match on the `pattern_sast`.